frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage.tsx

```diff
@@ -0,0 +1,194 @@
+      type: 'backingStorage/setExternalProvider',
+      payload: value,
+    });
+
+  return (
+    <FormSelect
+      aria-label={t('ceph-storage-plugin~Select external system from list')}
+      value={selectedProvider}
+      className="odf-backing-storage__selection--width"
+      onChange={handleSelection}
+    >
+      {hasOCS && <FormSelectOption key={RHCS} value={RHCS} label="Red Hat Ceph Storage" />}
+      {providersList.map((p) => (
+        <FormSelectOption key={p.id} value={p.id} label={p.label} />
+      ))}
+    </FormSelect>
+  );
+};
+
+type ExternalSystemSelectionProps = {
+  dispatch: React.Dispatch<React.ReducerAction<WizardReducer>>;
+  selectedProvider: WizardState['backingStorage']['externalProvider'];
+  providersList: ExternalProvider[];
+  hasOCS: boolean;
+};
+
+const StorageClassSelection: React.FC<StorageClassSelectionProps> = ({ dispatch, selected }) => {
+  const onStorageClassSelect = (sc: StorageClassResourceKind) =>
+    dispatch({
+      type: 'wizard/setStorageClass',
+      payload: { name: sc?.metadata?.name, provisioner: sc?.provisioner },
+    });
+  return (
+    <div className="odf-backing-storage__selection--width">
+      <StorageClassDropdown
+        noSelection
+        onChange={onStorageClassSelect}
+        selectedKey={selected.name}
+        filter={filterSCWithoutNoProv}
+        data-test="storage-class-dropdown"
+      />
+    </div>
+  );
+};
+
+type StorageClassSelectionProps = {
+  dispatch: WizardDispatch;
+  selected: WizardState['storageClass'];
+};
+
+const formatStorageSystemList = (storageSystems: StorageSystemKind[] = []): StorageSystemSet =>
+  storageSystems.reduce(
+    (kinds: StorageSystemSet, ss: StorageSystemKind) => kinds.add(ss.spec.kind),
+    new Set(),
+  );
+
+type StorageSystemSet = Set<StorageSystemKind['spec']['kind']>;
+
+export const BackingStorage: React.FC<BackingStorageProps> = ({
+  state,
+  dispatch,
+  storageSystems,
+  storageClass,
+  error,
+  loaded,
+}) => {
+  const { t } = useTranslation();
+
+  const formattedSS: StorageSystemSet = formatStorageSystemList(storageSystems);
+
+  const hasOCS: boolean = formattedSS.has(StorageClusterIdentifier);
+
+  const externalProviders = ODF_EXTERNAL_PROVIDERS.filter(
+    (provider) => !formattedSS.has(provider.kind),
+  );
+
+  React.useEffect(() => {
+    /*
+     Allow pre selecting the "external connection" option instead of the "existing" option 
+     if an OCS Storage System is already created and no external system is created.
+    */
+    if (hasOCS && externalProviders.length) {
+      dispatch({ type: 'backingStorage/setType', payload: BackingStorageType.EXTERNAL });
+    }
+  }, [dispatch, externalProviders.length, hasOCS]);
+
+  const { type, externalProvider } = state;
+
+  const showExternalSystemSelection = type === BackingStorageType.EXTERNAL;
+  const showStorageClassSelection = !hasOCS && type === BackingStorageType.EXISTING;
+  const RADIO_GROUP_NAME = 'backing-storage-radio-group';
```

Comment:
Move it above useEffect

Reply:
The reason to keep it here is that the logic above hook is required for hook.